cEP-0016.md

```diff
@@ -0,0 +1,231 @@
+   commits, `git revert` , `git merge` and commits that disable ci builds.
+2. Each type of commit is then handled as a separate case.
+3. For travis commits:
+   Travis commits are unseen merge commits that are send to CI when GitHub merge
+   commits are added to 3+ neatly stacked PRs. These merge commits are different
+   as they are not made using `git merge`. Instead they are made by Github
+   automatically when a PR is commited by more than two people. GitCommitBear at
+   present does not identify such commits and thus fails. This failure is then
+   reflected in travis builds causing it to fail as well.
+   - As a solution to this problem, settings will be addded in GitCommitBear that
+     would identify travis merge commits and ignore them. Therefore, the
+     GitCommitBear will run without failing and so will the travis build.
+4. For commits that disable ci builds:
+   coala uses Travis CI, Circle CI, AppVeyor, GitLab CI and Semaphore CI for
+   continuous integration. Most of these support `[ci skip]` and `[skip ci]` for
+   skipping a build for a particular commit. Commits that have `[ci skip]` or
+   `[skip ci]` anywhere in the commit messages are ignored by them. With this
+   project, the GitCommitBear would support skipping builds for CI engines that
+   use these special sequences for skipping jobs. For others that do not support
+   such commands, the bear would give an appropriate message.
+   - Commit contents are checked to find `[ci skip]`.
+   - For such commits, it is checked that the commit body explains a valid
+     reason why this change should not include ci builds.
+   - Files modified in the commit should include files not covered by `.coafile`
+     or tests. If this is not the case, an error is raised to exclude `[ci skip]`
+     from the commit message.
+5. For merge commits:
+   - Merge commits are identified by checking the parents of the commit.
+   - In case the bear finds that the commit tries to merge master onto the pull
+     request, message is displayed to perform a `git rebase` instead.
+6. For revert commits:
+   - Revert commits are identified based on the format of the commit.
+   - For revert commits, the patches of both commits are compared to ensure that
+     they are in fact reverse of each other. Minor differences are allowed.
+   - In case of total mismatch, a note in the commit message highlighting that
+     its not a clean revert is expected.
+   - A message suggesting improvements in the commit message is displayed.
+
+## Code Prototypes
+
+A prototype for identifying travis merge commits:
+
+```python
+    if len(shortlog) > shortlog_length:
+        magic_merge_commit_regex = (^Merge [0-9a-f]+ into [0-9a-f]+$|....)
+
+        match = re.fullmatch(megic_merge_commit_regex, shortlog)
+        if match:
+            # ignore this commit from any further inspection
+```
+
+A prototype for identifying merge commits:
+
+```python
+    def is_merge_commit(self, shortlog, body):
```

Comment:
Same like https://github.com/coala/cEPs/pull/114/files#r187790446

I think you meant to return a `bool` from here.